### PR TITLE
Replace session token API with PAT Lifecycle Management API

### PR DESCRIPTION
**Pull request by amorap**
- Replace the legacy session token API with the PAT Lifecycle Management API which is the recommendation.
- Fix a mismatch issue between configuration and code.

### CODE REVIEW

File: NonInteractivePatGenerationSample/README.md [hunk 1]
```diff
@@ -1,22 +1,39 @@
...
+
+From a shell or command line:
 ```
+git clone https://github.com/microsoft/azure-devops-auth-samples.git
+```
+
+### Step 2: Configure the sample to use your Azure AD application
 
-## Step 2: Run the sample
+Update the configuration file `App.config` with the information about your AAD application, AAD crendentials and Azure DevOps organization.
```
Comment (anydot) [inline NonInteractivePatGenerationSample/README.md:25]: typo: crendentials

File: NonInteractivePatGenerationSample/NonInteractivePatGenerationSample/Program.cs [hunk 2]
```diff
@@ -41,21 +42,35 @@ public static void Main(string[] args)
...
-            var sessionToken = new SessionToken
+            CreatePatUsingNewPatApi(connection);
+        }
+
+        /// <summary>
+        /// Creates a PAT using the PAT Lifecycle Management API (https://learn.microsoft.com/en-us/rest/api/azure/devops/tokens)
+        /// </summary>
+        /// <param name="connection">Azure DevOps connection</param>
+        /// <returns></returns>
+        private static void CreatePatUsingNewPatApi(VssConnection connection)
```
Comment (anydot) [inline NonInteractivePatGenerationSample/NonInteractivePatGenerationSample/Program.cs:54]: Why not just "CreatePat"? By now there's nothing new about this API anyway ;)

**Review by anydot — APPROVED** [review]

File: NonInteractivePatGenerationSample/NonInteractivePatGenerationSample/Program.cs (hunk 2 above)
Comment (amorap) [reply NonInteractivePatGenerationSample/NonInteractivePatGenerationSample/Program.cs]: Changed. Originally I was thinking of leaving the old sample for history purposes.

File: NonInteractivePatGenerationSample/NonInteractivePatGenerationSample/Program.cs [hunk 3]
```diff
@@ -41,21 +42,35 @@ public static void Main(string[] args)
...
-                Scope = "vso.work"
+                Scope = "vso.work vso.graph"
             };
 
-            var pat = client.CreateSessionTokenAsync(
-                sessionToken,
-                tokenType: SessionTokenType.Compact,
-                isPublic: false
-                ).Result;
+            var patTokenResult = client.CreatePatAsync(request).Result;
```
[the remark this answers is not in the document]
Comment (amorap) [reply NonInteractivePatGenerationSample/NonInteractivePatGenerationSample/Program.cs]: Good point. I've updated the code.

**Pull request merged**